library/core/src/fmt/num.rs

```diff
@@ -480,7 +476,7 @@ impl_Exp!(i128, u128 as u128 via to_u128 named exp_u128);
 
 /// Helper function for writing a u64 into `buf` going from last to first, with `curr`.
 fn parse_u64_into<const N: usize>(mut n: u64, buf: &mut [MaybeUninit<u8>; N], curr: &mut usize) {
-    let buf_ptr = MaybeUninit::slice_as_mut_ptr(buf);
+    let buf_ptr = buf.as_mut_ptr().inner();
```

Comment:
This could potentially be replaced with a bunch of `MaybeUninit::write`s, but there are so many of them that I don't want to risk bounds checking perf.

Reply:
`add` is already unsafe, so I guess probs not needed.